### ISSUE TITLE
Allow final column in fixed-width files to be variable width

### ISSUE THREAD

**Issue by geekcyclist** [jehugaleahsa/FlatFiles#63]
**Problem I Face**

I have been assigned the task of modernizing a number of batch/console apps that parse fixed width files. One quirk is that the final column before the End of Line (EOL) in these files is **variable width**, typically because it contains an **encoded image** or similar variable and often lengthy content.

The current applications take one of two approaches:

1. The oldest read each line of the file 'manually' in a loop, and chunk off each column value using basic string parsing until the last column and then just reads to EOL.
2. The others use the Microsoft.VisualBasic.FileIO.TextFieldParser. This class allows you to specify an array of integer field widths similar to the mapping of window values (but not nearly as elegant or readable). In that class a declaration like this: 

```vb
Using parser As New Microsoft.VisualBasic.FileIO.TextFieldParser("C:\sample.txt")
    parser.TextFieldType = Microsoft.VisualBasic.FileIO.FieldType.FixedWidth
    parser.FieldWidths = {10, 20, 20, -1}
```
the "-1" in the final position of the FieldWidths array allows the final column to be variable width.

`FlatFiles` does not appear to support variable length final columns in Fixed-Width files.

**Solution Ideas**

Because the syntax of the `FixedLengthTypeMapper` is so elegant, I would love to see a solution like:
```csharp
public List<Customer> Parse(StringReader stringReader)
{
    var mapper = FixedLengthTypeMapper.Define<Customer>();
    mapper.Property(c => c.Id, new Window(10)).ColumnName(nameof(Customer.Id));
    mapper.Property(c => c.FirstName, new Window(20)).ColumnName(nameof(Customer.FirstName));
    mapper.Property(c => c.LastName, new Window(20)).ColumnName(nameof(Customer.LastName));
    mapper.Property(c => c.Image, new Window(-1)).ColumnName(nameof(Customer.Image));
    
   var customers = mapper.Read(stringReader, null).ToList();
   return customers;
 }
```
where passing -1 to the `Window` allows it to read to EOL.

Other possibilities I thought that might be clearer than relying on the "-1" *magic value* would be to have a `VariableWindow`, `LastWindow` or 'EndOfLineWindow` that would still work with the mapping, not take a width, and be expressive of the variable nature of the last column in the file.

**Alternative Attempts and Workarounds**

I attempted to pass -1, 0 and a length much longer than the final column to the Window() and all threw expected exceptions based on the current design.

I attempted to switch to using a `FixedLengthSchema` but ran into the same issue that a `Window` is required by `AddColumn`.

I could continue to use ` Microsoft.VisualBasic.FileIO.TextFieldParser` which appears to be receiving ongoing support, but the syntax in `FlatFiles` is much better.

**Comment by jehugaleahsa** [comment]
This seems like a cool idea. For now you could probably listen for the
RecordRead event and grab anything past the past column's position. I'm
going to see if I add some sort of setting to the schema object.

**Comment by jehugaleahsa** [comment]
I just pushed version 4.7.0. Please read the [CHANGELOG](https://github.com/jehugaleahsa/FlatFiles/blob/master/CHANGELOG.md#470-2020-02-15) to see the changes I made. For an example, take a look at these unit tests: 
https://github.com/jehugaleahsa/FlatFiles/blob/master/FlatFiles.Test/FixedLengthReaderTester.cs#L608
https://github.com/jehugaleahsa/FlatFiles/blob/master/FlatFiles.Test/FixedLengthReaderTester.cs#L655

Basically, rather than supporting `-1` explicitly, I have a `Window.Trailing` singleton instance that I used to detect this situation. Same idea, though. Please give this a try and let me know if you encounter any issues.

**Comment by geekcyclist** [comment]
I will take a look tonight or tomorrow. I'm excited. I had just spent a bit
looking at my fork of the repo and wasn't at all sure how I would have
implemented this yet.

Gary

**Comment by jehugaleahsa** [comment]
Closing for now. Feel free to reopen if you run into any problems.
